### ISSUE TITLE
关于用MNN中python的训练量化完成文本识别任务的post量化问题

### ISSUE THREAD

**Issue by Danee-wawawa** [alibaba/MNN#949]
您好，我在做文本识别任务，我想用MNN中python的训练量化完成post量化，在修改代码时有一个地方有个疑问，如下：

https://github.com/alibaba/MNN/blob/master/pymnn/examples/MNNTrain/quantization_aware_training/imagenet_dataset.py#L87

因为我是文本识别任务，所以这个地方的”dl“不是类别数字，而是文本内容的字符串，应该如何修改呢？
希望可以指导下，谢谢。

**Comment by yyfcc17** [comment]
你这些字符串要拿来训练也要编码吧，你直接传编码之后的内容过去

**Comment by dandanli814** [comment]
@yyfcc17 
好的，谢谢，我传下看看。

**Comment by dandanli814** [comment]
@yyfcc17 
您好，我在实现的时候发现，不同图像的文本内容编码后的序列长度不一样，这会导致每一个batch里面的label长度都不一样，这种要如何解决呢，因为我是把下面这行
https://github.com/alibaba/MNN/blob/master/pymnn/examples/MNNTrain/quantization_aware_training/imagenet_dataset.py#L87
改成以下：
dl = F.const(text, [len(text)], F.data_format.NHWC, F.dtype.int)
不知道这样是否正确？（text即为文本编码后的序列，比如[2655, 3631, 2734, 930, 1339, 1061, 3498, 108]）

**Comment by yyfcc17** [comment]
你可以试试，batch size > 1可能会出错

**Comment by dandanli814** [comment]
@yyfcc17 
好的，我试下，谢谢。
另外我想问下利用python的训练量化如何实现让每个batch内图像大小一样，每个batch之间图像大小不一样，MNN提供的https://github.com/alibaba/MNN/blob/master/pymnn/examples/MNNTrain/quantization_aware_training/quant_aware_training.py这个函数中主要是以下语句实现DataLoader：

    train_dataset = ImagenetDataset(image_path, val_txt, True)
    test_dataset = ImagenetDataset(image_path, val_txt, False)
    train_dataloader = MNN.data.DataLoader(train_dataset, batch_size=32, shuffle=True)
    test_dataloader = MNN.data.DataLoader(test_dataset, batch_size=10, shuffle=False)

我看了下MNN.data.DataLoader，它里面貌似没有类似Pytorch的torch.utils.data.DataLoader中的collate_fn这样的参数以实现上述功能。那该如何实现呢？还麻烦您指导下。谢谢。
